### PR TITLE
build: add gh workflow for building releases

### PR DESCRIPTION
**Pull request by macintoshpie**
#### What does this PR do?
Add a github workflow that's triggered on tag pushes, which builds the webpage along with the data dict stuff and adds it to the release.
The docs html is now being built with a forked version of [this repo](https://github.com/bitfehler/xs3p). It is now much smaller in size and faster.

Also deletes the json schema b/c we aren't using it. Also deletes enumerations.json and DataDictionary.xlsx b/c these are "built" files which are included in the release.

#### How should this be manually tested?
Once merged, push a test tag (non pre-release) and verify:
1. it created the release
2. it included index.html and json and xlsx files
3. the description of the release is the most recent section of the changelog

Then try it again with a pre-release tag (ie include vx.x.x-prx) and verify that it labels the release as a pre-release

#### What are the relevant tickets?
#305 

#### Screenshots (if appropriate)
Screen Shot 2021-04-15 at 8 28 46 AM
Screen Shot 2021-04-15 at 8 28 33 AM

### CODE REVIEW

**Review by nllong — APPROVED** [review]
This is great, thanks for automating as much as we can here. Few comments:

1) I'm okay deprecating the json schema. We should add a note to the buildingsync.net website that it is deprecated though.
2) I don't really remember why we were committing the DataDictionary and enumerations.json file. The files may be linked to from the BuildingSync.net website (maybe??).
3) I really like that we are ridding ourselves of the proprietary/for-purchase software!

💥🤜🤛

**Comment by nllong** [comment]
I build the schema doc locally and it looks okay. I really miss the graphical view, but it loaded really slow (if at all). Plus we should be encouraging the users to go to the selectiontool schema viewer (and eventually the new website).

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+          ruby-version: 2.7.2
+          bundler-cache: true
+      -
+        name: Test
+        run: bundle exec rake
+      -
+        name: Checkout xs3p
+        uses: actions/checkout@v2
+        with:
+          repository: macintoshpie/xs3p
```
[the remark this answers is not in the document]
Comment (macintoshpie) [reply .github/workflows/publish.yml:25]: I think that's a good idea, I'll create a repo

File: docs/release_instructions.md [hunk 2]
```diff
@@ -2,46 +2,47 @@
...
+git push origin 
+```
+
+This should trigger a GitHub workflow for building and publishing the release. Once the release has been successfully published on GitHub, continue.
+
+### Update BuildingSync Website
+
+At this point the GitHub action for publishing the release should be finished. Now we need to update the docs/data in [this repo](https://github.com/BuildingSync/website]). Check out the repo and make a new branch.
+
+* Copy `index.html` from the release into `schema/vX.Y.Z/documentation/index.html`.
```
[the remark this answers is not in the document]
Comment (macintoshpie) [reply docs/release_instructions.md:42]: I agree, I think we will need to create a new action for the site repo to just pull the release and its assets and auto copy/build the pages it needs

**Pull request merged**